lib/client.js

```diff
@@ -256,7 +264,10 @@ class Client extends Dispatcher {
     }
 
     try {
-      const request = new Request(opts, handler)
+      console.log('Executing request')
+      const request = this[kMaxRedirections] > 0
+        ? new Request(opts, new RedirectHandler(this, { ...opts, origin: this[kUrl] }, handler))
+        : new Request(opts, handler)
```

Comment:
cc: @ronag 
I would like to know if I'm following the correct path as when trying to test it, I'm falling into the following error:

```text
error; ClientClosedError: The client is closed
    at Client.dispatch (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/client.js:277:15)
    at RedirectHandler.onComplete (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/handler/redirect.js:101:23)
    at Request.onComplete (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/core/request.js:159:27)
    at Parser.onMessageComplete (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/client.js:837:15)
    at wasm_on_message_complete (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/client.js:430:28)
    at null.<anonymous> (wasm://wasm/0002924a:0:1998)
    at null.<anonymous> (wasm://wasm/0002924a:0:6386)
    at null.<anonymous> (wasm://wasm/0002924a:0:3006)
    at llhttp_execute (wasm://wasm/0002924a:0:1055)
    at Parser.execute (/Users/carlosgerardofuentesrios/Documents/Workspace/MetCoder95/undici/lib/client.js:528:32) {
    code: 'UND_ERR_CLOSED'
}
```

Reply:
Any help? 😅 
I'm running out of options, not sure what's causing the client to be `closed` when the `RedirectHandler` tries to set dispatch the new request. Following the stack trace I only see the happy path once the request is done, but ran out of ideas what's closing the client

Reply:
Found that the error is mostly in the testing part, I'll continue with it soon 👍